src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -19,10 +19,6 @@ internal class HtmlMarkupParser : TokenizerBackedParser<HtmlTokenizer>
             SyntaxFactory.Token(SyntaxKind.Bang, "!"),
```

Comment:
Not blocking the PR on this but this class definitely need to be refactored in ways like CSharpCodeParser was. CSharpCodeParser is beaaautiful in terms of its factorings and this could benefit from similar treatment.

Reply:
Yup, that is the goal 🏁